plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/PlayAssetDelivery.java

```diff
@@ -20,15 +20,81 @@
 
 import org.godotengine.godot.Godot;
 import org.godotengine.godot.plugin.GodotPlugin;
+import org.godotengine.godot.plugin.SignalInfo;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
+
+/**
+ * This class is served as a middleware, to expose the Play Core Java library to the Godot runtime,
+ * so Godot Play Asset Delivery front-end API is able to access the related Java APIs
+ * This class is intended to be compile to an .AAR Android Library, which can be loaded into
+ * Godot Engine as an Godot Android Plugin
+ */
 public class PlayAssetDelivery extends GodotPlugin {
+
     public PlayAssetDelivery(Godot godot) {
         super(godot);
     }
 
     @NonNull
     @Override
     public String getPluginName() {
-        throw new UnsupportedOperationException("Method not implemented yet!");
+        return "PlayAssetDelivery";
+    }
+
+    /**
+     *
+     * @return a list of all the available plugin method names that the Godot runtime can call
+     */
+    @NonNull
+    @Override
+    public List<String> getPluginMethods() {
+        return Arrays.asList("cancel", "fetch", "getAssetLocation", "getPackLocation",
+                "getPackLocations", "getPackStates", "removePack", "showCellularDataConfirmation");
+    }
+
+    /**
+     *
+     * @return a set containing all the signals the Godot runtime is able to receive
+     * AssetPackStateUpdateSignal - passes AssetPackState serialized as Dictionary
+     * All the signals below also passes signalID
+     * fetchSuccess - passes AssetPackStates serialized as Dictionary
+     * fetchError - passes Error serialized as Dictionary
+     * getPackStatesSuccess - passes AssetPackStates serialized as Dictionary
+     * getPackStatesError - passes Error serialized as Dictionary
+     * removePackSuccess - passes name of the pack removed
+     * removePackError - passes name of the pack to be removed along with the Error serialized
+     * as Dictionary
+     * showCellularDataConfirmationSuccess - passes Integer indicating how the user responded to
+     * the dialog
+     * showCellularDataConfirmationError - passes Error serialized as Dictionary
+     */
+    @NonNull
+    @Override
+    public Set<SignalInfo> getPluginSignals() {
+        Set<SignalInfo> availableSignals = new HashSet<>();
+        availableSignals.add(new SignalInfo("assetPackStateUpdateSignal",
```

Comment:
I'm not sure if adding these line breaks is the best practice, since these lines are just little bit over 80 characters.

Reply:
Google's java style guide uses 100 characters per line, so you can use that convention if you prefer. If they're still over then line breaks are indeed best practice.

Reply:
My bad, I just double checked my Android Studio and the line limit is set at 100 characters indeed.